lib/index_finder.rb

```diff
@@ -0,0 +1,20 @@
+# Hides some of the dirtyness in selecting/generating the correct index objects
+class IndexFinder
```

Comment:
👍  Extracting this makes a lot of sense.

General ruby style question: When a class is not meant to be instantiated, is there any advantage to using a `class` rather than a `module` for defining your methods on?